src/main/java/org/nrg/containers/api/DockerControlApi.java

```diff
@@ -302,6 +304,17 @@ public Container createContainerOrSwarmService(final ResolvedCommand resolvedCom
                 resolvedCommand.workingDirectory() :
                 null;
 
+        // let resource constraints default to 0, so they're ignored by Docker
```

Comment:
I think these lines may not be necessary. Leaving these properties as `null` would also cause docker to ignore them.

Reply:
If I pass resolvedCommand.reserveMemory() and the other two directly to createService, some tests fail... Should I investigate why? Doesn't hurt to keep this in my opinion.

Reply:
I thought they would be equivalent, but if you know that they are not equivalent and that this way works, you can leave it.

Reply:
I have very little experience with Java, so I don't "know" that they're not equivalent, the tests just happened to fail one way. Maybe it's because of the final declaration?